packages/next/src/next-request-context.ts

```diff
@@ -0,0 +1,60 @@
+/*
+WARNING: WATCH OUT IF YOU'RE MODIFYING THIS FILE!
+The types here must be kept in sync (or at least backwards-compatible)
+with "@next/request-context" in the Next.js codebase.
+
+This module defines `globalThis[Symbol.for("@next/request-context")]`
+which Next.js uses to access `waitUntil`.
+It is expected that the platform provides this.
+*/
+
+import { AsyncLocalStorage } from 'async_hooks';
```

Comment:
Should the be using the global `AsyncLocalStorage` instance if available similar to what we do in Next.js?

Reply:
hmmmm well we're doing a `require('async_hooks')` there anyway
https://github.com/vercel/next.js/blob/626b7f9de94a82e0721df9e9a29a5af850dc8ace/packages/next/src/server/node-environment-baseline.ts#L4-L9
so i think it's pretty much equivalent, don't see any obvious benefits from doing `globalThis.AsyncLocalStorage` here